drivers: serial: lpuart: Add configurable init priority

nordic-krch opened the pull request:
Low power uart depends on gpio and uart devices and it may
be used by other device initialized at the same level
(POST_KERNEL). Adding configurable init priority to allow
influencing initialization order.

Signed-off-by: Krzysztof Chruscinski <krzysztof.chruscinski@nordicsemi.no>

anangl commented on drivers/serial/Kconfig:
```diff
@@ -13,6 +13,16 @@ config NRF_SW_LPUART
+	int "Initialization priority"
+	default 45
+	help
+	  Initialization priority within POST_KERNEL level. It should be set to
+	  a value that ensures that device is initialized later than gpio and
+	  uart devices used by this device. Default is set to be between
```
How about adding a build assertion that will check if this condition (and also the priority value is lower than `KERNEL_INIT_PRIORITY_DEVICE` which is used by `bt_uart_init()` in h4.c) is met?

anangl replied:
Abandoned the idea after offline discussion.

nordic-krch replied:
We talked offline and decided to not add assert for now.

nordic-krch commented:
@anangl can you take another look as i added commits which change gpiote priority level (as lpuart expects that uarte and gpiote are on the same priority). Added also assert in lpuart to check that.

anangl commented on drivers/serial/uart_nrf_sw_lpuart.c:
```diff
@@ -1026,7 +1026,11 @@ static const struct uart_driver_api lpuart_api = {
 
-DEVICE_DEFINE(lpuart, "LPUART", lpuart_init, device_pm_control_nop,
+BUILD_ASSERT(DT_IRQ(DT_INST(0, nordic_nrf_gpiote), priority) ==
+	     DT_IRQ(DT_PARENT(DT_INST(0, nordic_nrf_sw_lpuart)), priority),
```
```suggestion
	     DT_IRQ(DT_PARENT(DT_NODELABEL(lpuart)), priority),
```
for consistency with: https://github.com/nrfconnect/sdk-nrf/blob/0575cab37f7569d8a8feed89d5ddf4545b212be5/drivers/serial/uart_nrf_sw_lpuart.c#L1033
Perhaps it would be better to also use `DT_NODELABEL(gpiote)` instead of `DT_INST(0, nordic_nrf_gpiote)` above.
